### PR TITLE
Fix CI by downgrading to cargo insta 1.29.0

### PR DESCRIPTION
**Pull request by konstin**
Since the (implicit) update to cargo-insta 1.30, CI would pass even when the tests failed. This downgrades to cargo insta 1.29.0 and CI fails again when it should (which i can't show here, because CI needs to pass to merge this PR). I've improved the unreferenced snapshot handling in the process

See https://github.com/mitsuhiko/insta/issues/392

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -54,21 +54,19 @@ jobs:
...
-      - run: cargo install cargo-insta
+      # cargo insta 1.30.0 fails for some reason
+      - run: cargo install cargo-insta@=1.29.0
       - run: pip install black[d]==23.1.0
       - name: "Run tests (Ubuntu)"
         if: ${{ matrix.os == 'ubuntu-latest' }}
-        run: |
-          cargo insta test --all --all-features --delete-unreferenced-snapshots
-          git diff --exit-code
+        run: cargo insta test --all --all-features --unreferenced reject
```
Comment (konstin) [inline .github/workflows/ci.yaml:62]: i've fixed the unreferenced snapshot situation while i was already at it

**Comment by konstin** [comment]
I've confirmed that this indeed catches CI failures again because main had an outdated schema

**Comment by zanieb** [comment]
Interesting, what's different in insta 1.30?

**Comment by konstin** [comment]
i don't know, and unfortunately i can't reproduce locally. I scrolled through the diff and https://github.com/mitsuhiko/insta/compare/1.29.0...1.30.0#diff-5174eb15293f46eb47c52ac6d42dce9f2f5dd7f8dca11068a2159b342e5c712cR615 is the only thing that looks possibly relevant

**Review by MichaReiser — APPROVED** [review]

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5589** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5589" target="_blank">Graphite</a>  👈
    * **PR #5612** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5612" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5589?utm_source=stack-comment).

**Pull request merged**